lib/jenkins_pipeline_builder/extensions/builders.rb

```diff
@@ -26,36 +26,90 @@
-              if job[:config]
-                configs do
-                  if job[:config].key? :predefined_build_parameters
-                    send('hudson.plugins.parameterizedtrigger.PredefinedBuildParameters') do
-                      properties job[:config][:predefined_build_parameters]
+  version '0' do
+    xml do |params|
+      params[:phases].each do |name, content|
+        send('com.tikal.jenkins.plugins.multijob.MultiJobBuilder') do
+          phaseName name
+          phaseJobs do
+            content[:jobs].each do |job|
+              send('com.tikal.jenkins.plugins.multijob.PhaseJobsConfig') do
+                jobName job[:name]
+                currParams job[:current_params] || false
+                exposedSCM job[:exposed_scm] || false
+                if job[:config]
+                  configs do
+                    if job[:config].key? :predefined_build_parameters
+                      send('hudson.plugins.parameterizedtrigger.PredefinedBuildParameters') do
+                        properties job[:config][:predefined_build_parameters]
+                      end
+                    end
+                    if job[:config].key? :properties_file
+                      send('hudson.plugins.parameterizedtrigger.FileBuildParameters') do
+                        propertiesFile job[:config][:properties_file][:file]
+                        failTriggerOnMissing job[:config][:properties_file][:skip_if_missing] || 'false'
+                      end
                     end
                   end
-                  if job[:config].key? :properties_file
-                    send('hudson.plugins.parameterizedtrigger.FileBuildParameters') do
-                      propertiesFile job[:config][:properties_file][:file]
-                      failTriggerOnMissing job[:config][:properties_file][:skip_if_missing] || 'false'
+                end
+                killPhaseOnJobResultCondition job[:kill_phase_on] || 'FAILURE'
+              end
+            end
+          end
+          continuationCondition content[:continue_condition] || 'SUCCESSFUL'
+        end
+      end
+    end
+  end
+
+  version '1.27' do
+    xml do |params|
+      params[:phases].each do |name, content|
+        send('com.tikal.jenkins.plugins.multijob.MultiJobBuilder') do
+          phaseName name
+          phaseJobs do
+            content[:jobs].each do |job|
+              send('com.tikal.jenkins.plugins.multijob.PhaseJobsConfig') do
+                jobName job[:name]
+                currParams job[:current_params] || false
+                exposedSCM job[:exposed_scm] || false
+                disableJob job[:disable_job] || false
+                if maxRetries job[:max_retries]
+                  maxRetries job[:max_retries]
+                  enableRetryStrategy job[:enable_retry_strategy] || false
+                end
+                abortAllJob job[:abort_all_job] || false
+                if job[:condition].to_s.strip.empty? || !job[:condition]
```

Comment:
This should probably be done the other way around for short circuit evaluation

Reply:
Also, if you're going to use job[:condition] this many times, might be worth pulling into a variable